client/sandbox/react-nextjs/pages/play/linkunlink.tsx

```diff
@@ -0,0 +1,201 @@
+        <p className="mb-2">
+          Current User: {currentUser?.name || 'None'} (ID:{' '}
+          {currentUserId || 'N/A'})
+        </p>
+        <button
+          onClick={seedData}
+          className="bg-blue-500 text-white px-4 py-2 rounded mr-2"
+        >
+          Seed Data
+        </button>
+        <select
+          value={currentUserId}
+          onChange={(e) => setCurrentUserId(e.target.value)}
+          className="px-4 py-2 border rounded"
+        >
+          <option value="">Select User</option>
+          {users.map((user) => (
+            <option key={user.id} value={user.id}>
+              {user.name} ({user.id.slice(0, 8)}...)
+            </option>
+          ))}
+        </select>
+      </div>
+
+      {groups.map((group) => {
+        const isOwner = group.ownerId === currentUserId;
+        const memberIds = new Set(group.members?.map((m: any) => m.id) || []);
+
+        return (
+          <div key={group.id} className="mb-6 p-4 border rounded">
+            <h2 className="text-xl font-semibold mb-2">
+              {group.name} (Owner:{' '}
+              {users.find((u) => u.id === group.ownerId)?.name})
+            </h2>
+
+            <div className="mb-4">
+              <h3 className="font-semibold mb-2">Current Members:</h3>
+              {group.members?.map((member: any) => (
+                <div key={member.id} className="flex items-center gap-2 mb-1">
+                  <span>{member.name}</span>
+                  <button
+                    onClick={() => removeMember(group.id, member.id)}
+                    className="text-red-500 text-sm"
+                    title={
+                      isOwner
+                        ? 'Owner can remove anyone'
+                        : 'Users can only remove themselves'
+                    }
+                  >
+                    Remove{' '}
+                    {isOwner
+                      ? '(as owner)'
+                      : member.id === currentUserId
+                        ? '(self)'
+                        : '(no permission)'}
+                  </button>
+                </div>
+              ))}
+            </div>
+
+            <div>
+              <h3 className="font-semibold mb-2">Available Users:</h3>
+              {users
+                .filter((u) => !memberIds.has(u.id))
+                .map((user) => (
+                  <div key={user.id} className="flex items-center gap-2 mb-1">
+                    <span>{user.name}</span>
+                    <button
+                      onClick={() => addMember(group.id, user.id)}
+                      className="text-green-500 text-sm"
+                    >
+                      Add
+                    </button>
+                  </div>
+                ))}
+            </div>
+          </div>
+        );
+      })}
+    </div>
+  );
+}
+
+export default function Page() {
+  return <EphemeralAppPage schema={schema} perms={perms} Component={Example} />;
```

Comment:
cc @tonsky 

Tried playing with link/unlink, by creating this play page. I did the following steps: 

1. Clicked 'Seed', to add users and groups
2. Picked the 'current user' as an `Owner`. 
3. Tried to click 'add' on someone. 

I got a perms-pass failure, with a hint on `["groups" :object]` 

I am not sure if my example is wrong through. Do you mind checking?

Reply:
Ah! Nice catch, thanks @tonsky !